recipes/libftdi/1.x/conanfile.py

```diff
@@ -10,94 +13,86 @@ class LibFtdiConan(ConanFile):
             del self.options.fPIC
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-
-        self._cmake = CMake(self)
-        options = {
-            "BUILD_TESTS": False,
-            "EXAMPLES": False,
-            "FTDI_EEPROM": self.options.build_eeprom_tool,
-            "FTDIPP" : self.options.enable_cpp_wrapper,
-            "STATICLIBS": not self.options.shared,
-            "ENABLE_STREAMING": self.options.use_streaming,
-        }
-        self._cmake.definitions.update(options)
-        self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # Boolean values are preferred instead of "ON"/"OFF"
+        tc.variables["BUILD_TESTS"] = False
+        tc.variables["EXAMPLES"] = False
+        tc.variables["FTDI_EEPROM"] = self.options.build_eeprom_tool
+        tc.variables["FTDIPP"] = self.options.enable_cpp_wrapper
+        tc.variables["ENABLE_STREAMING"] = self.options.use_streaming
+        tc.variables["LIB_SUFFIX"] = ""
+        tc.generate()
+        # In case there are dependencies listed on requirements, CMakeDeps should be used
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def requirements(self):
         self.requires("libusb/1.0.24")
-        self.requires("boost/1.75.0")
+        if self.options.enable_cpp_wrapper:
+            self.requires("boost/1.75.0")
 
     def validate(self):
         if self.settings.compiler == "Visual Studio" and self.options.use_streaming:
             raise ConanInvalidConfiguration("VS doesn't not compile with enabled option use_streaming")
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
         lib_folder = os.path.join(self.package_folder, "lib",)
-        tools.rmdir(os.path.join(lib_folder, "cmake"))
-        tools.rmdir(os.path.join(lib_folder, "pkgconfig"))
+        rmdir(self, os.path.join(lib_folder, "cmake"))
+        rmdir(self, os.path.join(lib_folder, "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "LibFTDI1"
-        self.cpp_info.names["cmake_find_package_multi"] = "LibFTDI1"
-        self.cpp_info.names["pkgconfig"] = "libftdi1"
-
-        self.cpp_info.components["ftdi"].names["pkg_config"] = "libftdi1"
+        self.cpp_info.names['cmake_find_package'] = "LibFTDI1"
+        self.cpp_info.names['cmake_find_package_multi'] = "LibFTDI1"
```

Comment:
While the package does provide package config files, it doesn't seem to provide anything more than some CMake modules that set lots of variables, i.e. no CMake targets are provided. It looks like the CMake file name should be `LibFTDI1` and the target names left to the default, at least for the new generator.